Assets/Shopify/UIToolkit/Components/DataCache.cs

```diff
@@ -0,0 +1,60 @@
+ namespace Shopify.UIToolkit {
+    using UnityEngine;
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// An interface for defining a data cache.
+    /// </summary>
+    public interface DataCache<T> {
```

Comment:
Kinda nitpicky.... but what kind of cache stores something other than data? lol

Reply:
Heh true enough. I guess I was concerned about naming collisions but since we're within our namespace I don't think we'll have any issues.